packages/realm-react/example/App.tsx

```diff
@@ -15,114 +15,44 @@
 // limitations under the License.
 //
 ////////////////////////////////////////////////////////////////////////////
-import React, { useCallback, useMemo } from "react";
-import { SafeAreaView, View, StyleSheet } from "react-native";
 
-import TaskContext, { Task } from "./app/models/Task";
-import IntroText from "./app/components/IntroText";
-import AddTaskForm from "./app/components/AddTaskForm";
-import TaskList from "./app/components/TaskList";
-import colors from "./app/styles/colors";
+// 1. Move your App code to e.g. AppMain.tsx and create this outer App.tsx which is
+// responsible for logging in the user and setting up sync on the Realm.
 
-const { useRealm, useQuery, RealmProvider } = TaskContext;
+import React, { useEffect, useState } from "react";
+import { Realm } from "@realm/react";
 
-function App() {
-  const realm = useRealm();
-  const result = useQuery(Task);
+import TaskContext from "./app/models/Task";
+import { AppMain } from "./AppMain";
 
-  const tasks = useMemo(() => result.sorted("createdAt"), [result]);
+// 2. Add your Realm app details here, using whatever login mechanism you wish
+const app = new Realm.App({ id: "application-0-bzsvu" });
+const credentials = Realm.Credentials.anonymous();
 
-  const handleAddTask = useCallback(
-    (description: string): void => {
-      if (!description) {
-        return;
-      }
+const App = () => {
+  // 3. Store the logged in user in state
+  const [user, setUser] = useState<Realm.User | undefined>();
```

Comment:
This could be initialised with `app.currentUser`

Reply:
Just thinking about possible hooks.  We might be able to extract a lot of this into a `useAnonUser` hook:
```
const user = useAnonUser(app)
```
--or--
```
const {userAnonUser} = createAppHooks(app)
const App = () => {
const user = useAnonUser();
...
}
```

Reply:
It's also possible we will want an AppProvider...but perhaps just a generator function that returns hooks with an `Realm.App` as an argument would suffice.